src/main/java/org/opentripplanner/graph_builder/linking/SimpleStreetSplitter.java

```diff
@@ -184,14 +184,14 @@ public boolean link(Vertex vertex, TraverseMode traverseMode, RoutingRequest opt
             .collect(Collectors.toList());
 
         // Make a map of distances to all edges.
-        final TIntDoubleMap distances = new TIntDoubleHashMap();
+        final TObjectDoubleMap<Edge> distances = new TObjectDoubleHashMap<>();
         for (StreetEdge e : candidateEdges) {
```

Comment:
This relay on all Edge classes to have a propper hashCode and equals - do they?

Reply:
Edges are considered to have identity equality rather than semantic equality, i.e. any two separate edge instances are never equal even if they have the same characteristics. Identity equality is the default behavior of equals and hashCode on the Object base class. This should perhaps be stated clearly on the Javadoc of Edge, because identity equality does assume people will not override equals and hashcode on any classes in this hierarchy.